web-24wi/assignments/nathanMcL/StudentOriginatedSoftware/week2/Calculator_app/calculator.js

```diff
@@ -0,0 +1,103 @@
+// Calculator state variables
+let runningTotal = 0;
+let buffer = "0";
+let previousOperator;
+const screen = document.querySelector(".screen");
+
+// Handles the button clicks
+function buttonClick(value) {
+    if (isNaN(parseInt(value))) {
+        handleSymbol(value);  // Handle non-numeric symbols
+    } else {
+        handleNumber(value);  // Handle numeric values
+    }
+    rerender();  // Update the display
+}
+
+// Handles the number input
+function handleNumber(value) {
+    if (buffer === "0") {
+        buffer = value;  // Replace initial buffer value
+    } else {
+        buffer += value;  // Append number to buffer
+    }
+}
+
+// Handles the mathematical operations (+, -, x, ÷)
+function handleMath(value) {
+    if (buffer === "0") {
+        return;  // Exit if buffer is zero
+    }
+
+    const intBuffer = parseInt(buffer);  // Convert buffer to integer
+    if (runningTotal === 0) {
+        runningTotal = intBuffer;  // Set running total if it's the first operation
+    } else {
+        flushOperation(intBuffer);  // Perform the pending operation
+    }
+
+    previousOperator = value;  // Store current operation
```

Comment:
does value store a number, or an arithmetic operator?

Reply:
the value stores the current arithmetic operator